components/board/pi/impl/board.go

```diff
@@ -68,9 +70,10 @@ type piPigpio struct {
 }
 
 var (
-	pigpioInitialized bool
-	instanceMu        sync.Mutex
-	instances         = map[*piPigpio]struct{}{}
+	pigpioInitialized           bool
+	instanceMu                  sync.RWMutex
+	instances                   = map[*piPigpio]struct{}{}
+	instanceCtx, instanceCancel = context.WithCancel(context.TODO())
```

Comment:
is there a way to grab a parent context ? otherwise we should use Background

Reply:
The implementation before this PR also used `context.TODO()`, so the new version isn't any worse. If it's easy to change to a different context, that's fine. but if it becomes complicated, I think that should be a separate PR. Matt Vella's team is doing a workshop with NYU students this coming week, and they're hoping this PR will be merged by then so they can get accurate readings from the ultrasonic sensor.